libs/entry-components/theming-setup.md

```diff
@@ -0,0 +1,95 @@
+# Theming Setup Guide
+
+## Overview
+
+This guide explains how to set up your project to begin using theming. The Entry theme works hand-in-hand with [Angular Material](https://material.angular.io/). It allows you to configure only certain things for simplicity and quick setup.
+
+### Prerequisites
+
+1. Install required [`@enigmatry packages`](#required-packages)
+2. Including [`generator`](#include-styles-generator) for theming setup
+3. Remove Angular Material [`prebuilt themes`](#remove-prebuilt-themes)
+4. Check further steps [`here`](#theming-configuration-sections)
+
+## Required packages
+
+Ensure that the latest version of the[`@enigmatry packages`](#enigmatry-packages) are installed:
+
+- [`entry-components`](https://github.com/enigmatry/entry-angular-building-blocks/tree/master/libs/entry-components)
+- [`entry-form`](https://github.com/enigmatry/entry-angular-building-blocks/tree/master/libs/entry-form)
+
+### @enigmatry/entry-components
+
+Reusable Entry components for Angular material. It comes with the default Roboto font. There are plenty of components that this library includes: button, dialog, file-input, search-filter, spinner, table, and many more. <br>
+Visit components demo app to check what is currently supported [here](https://entry-demo.enigmatry.com/).
+
+#### Installation
+
+```ts
+npm install @enigmatry/entry-components
+```
+
+### @enigmatry/entry-form
+Form components and helper methods used by entry-code-generation.
+
+#### Installation
+
+```ts
+npm install @enigmatry/entry-form
+```
+
+Import the `EntryFormModule` in your `feature.module` or `shared.module`
+
+```typescript
+import { EntryFormModule } from '@enigmatry/entry-form';
```

Comment:
This is not really needed for theming AFAIK. It's indeed needed for JS part of functionality but that should be described in JS part of docs (since many more steps are needed for it to work). Here we need to focus on styles only.